### PR TITLE
Refractor plugin backend

### PR DESCRIPTION
**Pull request by botatooo**
Use server methods instead of directly hitting up the browser endpoint, makes it less janky.

### CODE REVIEW

**Review by AAGaming00 — CHANGES_REQUESTED** [review]
Maybe do this for the install method too?

**Comment by botatooo** [comment]
Plugin installation already utilizes server methods, no?

**Comment by botatooo** [comment]
Oh, I see what you mean.

**Comment by AAGaming00** [comment]
@WerWolv opinions on the formatting applied here?

**Review by AAGaming00 — APPROVED** [review]
Code-wise LGTM, still waiting on Wolv's opinion on the formatter. Will test soon.

**Comment by WerWolv** [comment]
Code-wise everything looks alright to me as well.

Formatting-wise, it partially feels like you just used an auto formatter that only allows a certain number of characters per line.
For me this makes it somewhat harder to read. Array indexes in my opinion should never be split up onto more than one lines. If the index value is too long, I'd write the result to a variable and then use that to index.
For function calls, if there are way too many parameters, I'll usually put each parameter on its own line and not just put the entire list of parameters on the next line.

**Comment by botatooo** [comment]
I'll take a note and do what I can in the next commit.

**Comment by botatooo** [comment]
> Code-wise everything looks alright to me as well.
> 
> Formatting-wise, it partially feels like you just used an auto formatter that only allows a certain number of characters per line. For me this makes it somewhat harder to read. Array indexes in my opinion should never be split up onto more than one lines. If the index value is too long, I'd write the result to a variable and then use that to index. For function calls, if there are way too many parameters, I'll usually put each parameter on its own line and not just put the entire list of parameters on the next line.

Did what I could manually, if it needs changes i will revisit it with a proper editor and not GitHub mobile.

**Review by TrainDoctor — APPROVED** [review]
This is good for merging.

**Review by Gerrickjohnson — APPROVED** [review]

**Comment by AAGaming00** [comment]
This needs to be adapted for recent changes.

**Comment by botatooo** [comment]
> This needs to be adapted for recent changes.

Conflicts have been resolved.

**Review by TrainDoctor — CHANGES_REQUESTED** [review]
Tested it on my deck via manual file replacement and works like a charm.
Did encounter an issue where the uninstall modal shows up but neither confirm or cancel are selected. Hitting the "A" button to confirm as if confirm was selected seems to press the uninstall button behind the modal and uninstall the plugin as if confirm was selected.
Once this issue is resolved, this will be good to merge.

**Review by TrainDoctor — APPROVED** [review]
With recent discoveries testing @pastaq's changes the uninstall prompt issues appear to be in the main codebase so we'll look into sorting that ourselves. I'll need another approving review before this can fully go ahead.

**Review by AAGaming00 — APPROVED** [review]
LGTM, will test when I have time.

**Review by AAGaming00 — APPROVED** [review]

**Review by AAGaming00 — APPROVED** [review]

File: backend/main.py [hunk 1]
```diff
@@ -80,9 +94,7 @@ def __init__(self) -> None:
         self.web_app.add_routes([get("/auth/token", self.get_auth_token)])
 
         for route in list(self.web_app.router.routes()):
-             self.cors.add(route)
```
Comment (TrainDoctor) [inline backend/main.py]: Why where these lines removed?

File: backend/utilities.py [hunk 2]
```diff
@@ -81,19 +81,15 @@ async def execute_in_tab(self, tab, run_async, code):
         try:
             result = await inject_to_tab(tab, code, run_async)
             if "exceptionDetails" in result["result"]:
-                return {
```
Comment (TrainDoctor) [inline backend/utilities.py:70]: Why where these lines removed?

**Review by TrainDoctor — CHANGES_REQUESTED** [review]
A few line removals that need to have their removal checked

**Comment by TrainDoctor** [comment]
> A few line removals that need to have their removal checked

Also a few errors relating to indentation and so on causing the build CI to fail.

**Review by AAGaming00 — APPROVED** [review]
Tested locally, everything seems to work.

**Pull request merged**